pkg/pillar/cmd/baseosmgr/baseosmgr.go

```diff
@@ -48,62 +41,43 @@ type baseOsMgrContext struct {
 	rebootImage        string    // Image from which the last reboot happened
 }
 
-var debug = false
-var debugOverride bool // From command line arg
-
-func Run(ps *pubsub.PubSub) {
-	versionPtr := flag.Bool("v", false, "Version")
-	debugPtr := flag.Bool("d", false, "Debug flag")
-	flag.Parse()
-	debug = *debugPtr
-	debugOverride = debug
-	if debugOverride {
-		log.SetLevel(log.DebugLevel)
-	} else {
-		log.SetLevel(log.InfoLevel)
-	}
-	if *versionPtr {
-		fmt.Printf("%s: %s\n", os.Args[0], Version)
-		return
-	}
-	agentlog.Init(agentName)
-	if err := pidfile.CheckAndCreatePidfile(agentName); err != nil {
-		log.Fatal(err)
-	}
+func newBaseOsMgrContext() *baseOsMgrContext {
+	baseOsMgrContext := baseOsMgrContext{}
+	baseOsMgrContext.agentBaseContext = agentbase.DefaultContext(agentName)
+	baseOsMgrContext.agentBaseContext.AgentOptions.SubscribeToGlobalConfig = true
+	return &baseOsMgrContext
+}
 
-	log.Infof("Starting %s\n", agentName)
+func (ctxPtr *baseOsMgrContext) AgentBaseContext() *agentbase.Context {
+	return &ctxPtr.agentBaseContext
+}
 
-	// Run a periodic timer so we always update StillRunning
-	stillRunning := time.NewTicker(25 * time.Second)
-	agentlog.StillRunning(agentName, warningTime, errorTime)
```

Comment:
Given that the event loop still needs to have code for stillrunning, moving these three lines to agentbase actually makes things a lot harder to understand. Unless we can figure out a way to dynamically plug in things into the event loop (there is a way; need a channel mux) it doesn't make sense to move this.